components/EventItem.tsx

```diff
@@ -0,0 +1,277 @@
+  const [numInvitations, setNumInvitations] = useState(0) // Number of requested invitations
+  const [isHours, setIsHours] = useState(true); // If time limit has >= 1 hour left
+  const [timeUntil, setTimeUntil] = useState(0); // Time (in either hours or minutes) until event starts
+  const [currentCapacity, setCurrentCapacity] = useState(1); // How many users have joined event
+  const [distance, setDistance] = useState(0); // Current distance user is from the event
+  const NUM_OF_LINES = 5;
+  const CAPACITY_PERCENTAGE = 0.75;
+
+  const setTime = () => {
+    if (startTime) {
+      const date = new Date()
+      const diffTime = Math.abs(startTime.getTime() - date.getTime());
+      const diffMin = Math.ceil(diffTime / (1000 * 60));
+      const diffHour = Math.ceil(diffTime / (1000 * 60 * 60));
+
+      if (diffMin < 60) {
+        setIsHours(false);
+        setTimeUntil(diffMin);
+
+      } else {
+        setIsHours(true);
+        setTimeUntil(diffHour);
+      }
+    } 
+  }
+
+  useEffect(() => {
+    setTime();
+  }, [startTime]);
+
+  const handleRequestToJoin = () => {
+    if (requested) {
+      setRequested(false);
+      setNumInvitations(numInvitations-1);
+    } else {
+      setRequested(true);
+      setNumInvitations(numInvitations+1);
+    }
+  };
+
+  return (
+    <View style={styles.secondaryContainerStyle}>
+      <View
+        style={[styles.spaceAround, styles.nestedReply]}
+      >
+      {/* Header (name, profile pic, event dot, distance) */}
+        <View
+          style={[styles.flexRow, {paddingBottom: '2%'}]}
+        >
+          <ProfileImageAndName
+            textStyle={{
+              fontWeight: writtenByYou ? "bold" : "normal",
+            }}
+            style={styles.profile}
+            userId={item.userId}
+          />
+          <IconButton
+              style={styles.eventDot}
+              iconName={"lens"}
+              size={15}
+              color={eventColor}
+              onPress={() => null}
+            />
+          <Text style={styles.distance}>{distance} mi</Text>
+        </View>
+
+        <Divider style={styles.divider}/>
+
+        {/* Description */}
+        <View style={styles.description}>
+          <Text numberOfLines={NUM_OF_LINES}
+            style={{
+              paddingHorizontal: '3%'
+            }}
+            accessibilityLabel={'description'}
+          >
+            {item.description}
+          </Text>
+        </View>
+
+        {/* Bottom Left Icons (time until event, max occupancy) */}
+        <View style={[styles.flexRow, {paddingBottom: '1%'}]}>
+          <View style={[styles.flexRow, {paddingLeft: '2%'}]}>
+            {startTime != null ?
+              <View style={{flexDirection: 'row'}} accessibilityLabel={'time until'}>
+                <IconButton
+                  iconName={"query-builder"}
+                  size={22}
+                  color={isHours ? 'grey' : 'red'}
+                  onPress={() => null}
```

Comment:
Using an iconButton for a non-interactive component would still give it a "button press" effect when the user taps it. Maybe we'll need another similar component that's a non-interactive version of the iconbutton, or edit the iconbutton to remove the touchableOpacity when onPress is undefined.